Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/SwapShiftController.cs

```diff
@@ -589,97 +589,66 @@ public static string GenerateNewGuid()
-                    // If the Retract request fails in Kronos.
-                    swapShiftResponse.Status = (int)HttpStatusCode.InternalServerError;
-                    swapShiftResponse.Body = new Body()
-                    {
-                        Error = new ResponseError
-                        {
-                            Code = Resource.KronosErrorStatus,
-                            Message = swapShiftRes?.Error.DetailErrors.Error.FirstOrDefault().Message,
-                        },
+        }
 
-                        ETag = null,
-                    };
-                    swapShiftResponse.Id = reqId;
+        public async Task<ShiftsIntegResponse> RetractOfferedShiftAsync(
+            SwapShiftMappingEntity map)
+        {
+            var allRequiredConfigurations = await this.utility.GetAllConfigurationsAsync().ConfigureAwait(false);
+            var swapShiftResponse = new ShiftsIntegResponse();
+            var user = map.RequestorKronosPersonNumber;
+
+            this.utility.SetQuerySpan(
+                Convert.ToBoolean(true, CultureInfo.InvariantCulture),
+                out var shiftStartDate,
+                out var shiftEndDate);
+
+            if (map.KronosStatus == ApiConstants.Submitted)
+            {
+                user = map.RequestedKronosPersonNumber;
+            }
+
+            if ((bool)allRequiredConfigurations?.IsAllSetUpExists)
+            {
+                var response = await this.swapShiftActivity.SubmitRetractionRequest(
+                    allRequiredConfigurations.KronosSession,
+                    map.KronosReqId,
+                    user,
+                    $"{shiftStartDate} - {shiftEndDate}",
+                    new Uri(allRequiredConfigurations.WfmEndPoint)).ConfigureAwait(false);
+
+                if (response?.Status == ApiConstants.Success)
+                {
+                    map.KronosStatus = ApiConstants.Retract;
+                    await this.swapShiftMappingEntityProvider.AddOrUpdateSwapShiftMappingAsync(map).ConfigureAwait(false);
+
+                    swapShiftResponse = new ShiftsIntegResponse()
+                    {
+                        Id = map.KronosReqId,
+                        Status = (int)HttpStatusCode.OK,
+                        Body = new Body()
+                        {
+                            Error = null,
+                            ETag = null,
+                        },
+                    };
+                }
+                else
+                {
+                    swapShiftResponse.Status = (int)HttpStatusCode.InternalServerError;
+                    swapShiftResponse.Body = new Body()
+                    {
+                        Error = new ResponseError
+                        {
+                            Code = Resource.KronosErrorStatus,
+                            Message = response.Error.Message,
+                        },
+
+                        ETag = null,
+                    };

```

Comment:
Can we use the methods in TeamsController to generate these responses and just return a bool to represent whether the api call was successful.